source/Tree/Tree.example.js

```diff
@@ -0,0 +1,238 @@
+
+        <ContentBoxParagraph>
+          <label className={styles.checkboxLabel}>
+            <input
+              aria-label="Close all nodes?"
+              checked={closeAll}
+              type="checkbox"
+              onChange={event => this.setState({closeAll: event.target.checked})}
+            />
+            Close all nodes?
+          </label>
+          <label className={styles.checkboxLabel}>
+            <input
+              aria-label="Use dynamic rows heights?"
+              checked={dynamicRowsHeights}
+              type="checkbox"
+              onChange={event => this.setState({dynamicRowsHeights: event.target.checked})}
+            />
+            Use dynamic rows heights?
+          </label>
+        </ContentBoxParagraph>
+
+        <InputRow>
+          <LabeledInput
+            label="Height"
+            name="height"
+            onChange={event => this.setState({height: parseInt(event.target.value, 10) || 1})}
+            value={height}
+          />
+          <LabeledInput
+            label="Scroll to"
+            name="scrollTo"
+            placeholder="Index..."
+            onChange={event => this.setState({scrollTo: parseInt(event.target.value, 10) || undefined})}
+            value={scrollTo}
+          />
+          <LabeledInput
+            label="Row height"
+            name="rowHeight"
+            onChange={event => this.setState({rowHeight: parseInt(event.target.value, 10) || 1})}
+            value={rowHeight}
+          />
+        </InputRow>
+
+        <div>
+          <AutoSizer disableHeight>
+            {({width}) => (
+              <Tree
+                height={height}
+                nodeGetter={this._nodeGetter}
+                overscanIndicesGetter={this.constructor.overscanIndicesGetter}
+                overscanRowCount={0}
+                ref={this._setRef}
+                rowHeight={rowHeight}
+                scrollToIndex={scrollTo}
+                width={width}
+              />
+            )}
+          </AutoSizer>
+        </div>
+      </ContentBox>
+    );
+  }
+
+  _createNode(depth = 0) {
+    const node = {
+      id: this.id,
+      name: `test-${this.id}`,
+      children: [],
+    };
+
+    this.id += 1;
+
+    if (depth === 5) {
+      return node;
+    }
+
+    for (let i = 0; i < 5; i++) {
+      node.children.push(this._createNode(depth + 1));
+    }
+
+    return node;
+  }
+
+  _createStates(value) {
+    const states = {};
+
+    for (const key of this.allKeys) {
+      states[key] = value;
+    }
+
+    return states;
+  }
+
+  * _nodeGetter() {
+    const {dynamicRowsHeights} = this.state;
+
+    const stack = [];
+
+    stack.push({
+      nestingLevel: 0,
+      node: this.root,
+    });
+
+    while (stack.length !== 0) {
+      const {node, nestingLevel} = stack.pop();
+
+      const isOpened = yield {
+        childrenCount: node.children.length,
+        height: dynamicRowsHeights ? getRandomInt(30, 60) : undefined,
```

Comment:
The dynamic height should probably come from the `context.list` items, each of which defines a `size`. It doesn't necessarily have to _be_ that size but it should be _based_ on that size so it doesn't change each time the function is called.